src/sshforwarding.rs

```diff
@@ -112,13 +112,19 @@ impl NetworkTunnel for SshForwarding {
             if line.starts_with("debug1:") {
                 tracing::debug!("ssh: {}", &line);
             } else if line.starts_with("Warning: Permanently added") {
-                tracing::debug!("ssh: {}", &line);
+                tracing::debug!("{}", &line);
             } else if line.contains("Permission denied") {
-                tracing::error!("ssh: {}", &line);
+                tracing::error!("{}", &line);
+                return Err(Error::SilentError)
             } else if line.contains("Network is unreachable") {
-                tracing::error!("ssh: {}", &line);
+                tracing::error!("{}", &line);
```

Comment:
I didn't realize that it was actually *this* process that was logging out via tracing::error!() in the first place. I thought it was ssh logging it directly, by inheriting stdout / stderr of this wrapping parent.

That changes my recommendation and I instead think it makes sense to skip the tracing::error lines and just bubble up structured errors which are the last output of this process.

Reply:
put differently, why aren't these returning errors ? How is it supposed to relate to TunnelExitNonZero ?